### ISSUE TITLE
16.12 regression: ON KEY trapping does not work

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#33]
Spin-off from issue #25. `ON KEY` trapping doesn't seem to work either with the up-arrow as in that example or with F1, so probably doesn't work at all.

**Comment by robhagemans** [comment]
```
10 ON KEY(1) GOSUB 1000
20 KEY(1) ON
30 GOTO 30
1000 PRINT "f1":RETURN
```

Pressing `F1` does not trigger the trap; rather, after `Ctrl+Break`, the `F1` macro is printed (`LIST`). Should trigger the trap and not print the macro.

**Comment by robhagemans** [comment]
fixed by commit 60b7bd6
